internal/plans/internal/planproto/planfile.proto

```diff
@@ -207,6 +212,36 @@ message OutputChange {
     bool sensitive = 3;
 }
 
+// ConditionType defines the type of condition block used to generate a result.
+enum ConditionType {
+	INVALID                = 0;
+	RESOURCE_PRECONDITION  = 1;
+	RESOURCE_POSTCONDITION = 2;
+	OUTPUT_PRECONDITION    = 3;
+}
+
+message ConditionResult {
+    // Address of the object containing the condition.
+    string addr = 1;
+
+    // Globally-unique address of the condition block.
+    string condition_addr = 2;
```

Comment:
This is more a note to me than anything else, since I just wanted to capture my thought process on this somewhere. :grimacing: 

Initially I was a little concerned about saving the implementation-detail address as part of this file format, but I guess it's okay because we do not document this file format as being for external consumption anyway, and it's okay that it's not durable between runs because a particular plan is _itself_ not durable between runs. The plan file has the configuration serialized along with this metadata and so there should be no situation where the condition IDs could get out of sync with the configuration.

I guess it could still be worth documenting in the comment here that this is a run-specific identifier and an implementation detail, if only for our own future reference in understanding what the intended use of this field is.

Reply:
Commented!